lib/node/index.js

```diff
@@ -743,7 +744,8 @@ Request.prototype.request = function(){
   }
 
   // query
-  this.query(url.query);
+  if (url.search)
+    this.query(url.search.substr(1));
```

Comment:
If there is a query string, pass it unmodified (skipping the leading `?` character).